engine/src/main/java/io/zeebe/engine/processor/ProcessingStateMachine.java

```diff
@@ -362,11 +367,20 @@ private void updateState() {
             LOG.error(ERROR_MESSAGE_UPDATE_STATE_FAILED, currentEvent, throwable);
             onError(throwable, this::updateState);
           } else {
+            notifyListener();
```

Comment:
Could we call this in the execute side effects method before calling the next read event?

And if this fails we do not want to retry yes?

Reply:
Is the backpressure data structure thread safe? How does this work with removing then command from another actor?

Reply:
> Is the backpressure data structure thread safe? How does this work with removing then command from another actor?

Yes. It is thread safe. It was already designed to be called from different actors: adding the command was done by commandApiMessageHandler, removing done by the processor when writing the response.

Reply:
> And if this fails we do not want to retry yes?
Should we retry? If any exception occurs here for back pressure notification, it won't be recoverable by retrying.

Reply:
Is then backpressure broken?

Reply:
I don't expect any exceptions here. If an exception occurs, then there is some serious bug :grinning: and it won't be recoverable by retrying.

Reply:
ok thanks